### DISCUSSION TITLE
https://xaviw.github.io/XaviDocs/%E5%B7%A5%E5%85%B7%E7%B3%BB%E5%88%97/%E7%BC%96%E8%BE%91%E5%99%A8/Windows+WSL%E6%90%AD%E5%BB%BA%E5%BC%80%E5%8F%91%E7%8E%AF%E5%A2%83.html

### DISCUSSION THREAD

**Discussion by giscus** [Xaviw/XaviDocs#46] [Announcements]
# https://xaviw.github.io/XaviDocs/%E5%B7%A5%E5%85%B7%E7%B3%BB%E5%88%97/%E7%BC%96%E8%BE%91%E5%99%A8/Windows+WSL%E6%90%AD%E5%BB%BA%E5%BC%80%E5%8F%91%E7%8E%AF%E5%A2%83.html

个人前端技术文档

https://xaviw.github.io/XaviDocs/%E5%B7%A5%E5%85%B7%E7%B3%BB%E5%88%97/%E7%BC%96%E8%BE%91%E5%99%A8/Windows+WSL%E6%90%AD%E5%BB%BA%E5%BC%80%E5%8F%91%E7%8E%AF%E5%A2%83.html

**Comment by Xaviw** [comment]
如果遇到：

**wsl: 检测到 localhost 代理配置，但未镜像到 WSL。NAT 模式下的 WSL 不支持 localhost 代理。**

可以在 `C:\Users\[username]` 目录下创建 `.wslconfig` 文件，写入:

```
[experimental]
autoMemoryReclaim=gradual
networkingMode=mirrored
dnsTunneling=true
firewall=true
autoProxy=true
```

然后在 PowerShell 中用 `wsl --shutdown` 命令关闭 WSL，再重启 Linux 终端

**Comment by Xaviw** [comment]
防火墙设置如果遇到：

**New-NetFirewallRule : The specified interface was not found on the system.**

可以在开始菜单搜索`防火墙`，进入`防火墙和网络保护`，选择`高级设置`，点击`Windows Defender 防火墙属性`，选择`公用配置文件`，点击`受保护的网络连接`后的`自定义`按钮，查看名称包含`vEthernet`和`WSL`的选项，用实际名称替换原命令中的`vEthernet (WSL)`
